src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,332 @@
+             .
+
+    >>> _is_latest_version_greater("23.8.0", "23.8.0")
+    Latest Version already installed
+    False
+    >>> _is_latest_version_greater("23.8.1", "24.0.0")
+    Latest Version already installed
+    False
+    >>> _is_latest_version_greater("23.8.1", "23.8.0")
+    True
+    >>> _is_latest_version_greater("24.0.0", "23.8.5")
+    True
+    >>> _is_latest_version_greater("invalid_version", "1.2.0")
+    Traceback (most recent call last):
+    ...
+    click.exceptions.ClickException: Invalid version format found
+    """
+    try:
+        logging.debug("Entering _compare_version")
+        latest_parts: Tuple[int, ...] = _parse_version(latest_version)
+        installed_parts: Tuple[int, ...] = _parse_version(installed_version)
+
+        if latest_parts > installed_parts:
+            return True
+        else:
+            print(f"Latest Version already installed")
+            return False
+    except ValueError  as e:
+        logging.info("Error: %s", e)
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'Invalid version format found') from e
+
+
+def _get_daqmx_installed_version() -> str:
+    """
+    Check for existing installation of DAQmx by checking for Version information.
+    Version information is read from the registry key.
+    
+    Returns:
+        str: The version of the installed NI-DAQmx if found, else an empty string.
+
+    >>> _get_daqmx_installed_version()  # doctest: +SKIP
+    ''
+    """
+    try:
+        logging.debug("Reading the registry entries to get installed DAQmx version")
+        with winreg.OpenKeyEx(
+            winreg.HKEY_LOCAL_MACHINE,
+            r"SOFTWARE\National Instruments\NI-DAQmx\CurrentVersion",
+            0,
+            winreg.KEY_READ | winreg.KEY_WOW64_32KEY,
+        ) as daqmx_reg_key:
+            product_name = winreg.QueryValueEx(daqmx_reg_key, "ProductName")[0]
+            product_version = winreg.QueryValueEx(daqmx_reg_key, "Version")[0]
+
+        if product_name == "NI-DAQmx":
+            logging.debug("Found registry entries for Product Name: %s and version %s", product_name , product_version )
+            return product_version
+    except FileNotFoundError:
+        logging.info("No existing NI-DAQmx installation found.")
+        logging.debug(traceback.format_exc())
+        return None
+    except PermissionError as e:
+        logging.info("Error: %s", e)
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'Permission denied while trying to read the registry key.') from e
+    except OSError as e:
+        logging.info("Error: %s", e)
+        logging.debug(traceback.format_exc())
+        raise click.ClickException(f'An OS error occurred while trying to read the registry key.') from e
+
+
+# Creating a temp file that we then close and yield - this is used to allow subprocesses to access
+@contextlib.contextmanager
+def multi_access_temp_file(*, suffix=".exe", delete=True):
```

Comment:
url_open seems to be able to handle a separate open file better. see example code [here](https://stackoverflow.com/questions/55864160/how-to-make-urllib-request-append-to-an-existing-file). consider that, and not creating a temp file, but not deleting. Then you can remove this function and use a simple with statement in the calling function.

Reply:
@zhindes, If we use url_open still we need to provide a filename as input and it needs to unique and we should also not leak the file. 
I had a offline discussion with @mshafer-NI he recommended this approach and we have tested and its working fine.
Do you still recommend to consider using url_open?
please let me know, I will look into that option

Reply:
I was suggesting something like this:

```python
with urllib.request.urlopen("ni.com/daqmx_installer") as response:
    with tempfile.NamedTemporaryFile(suffix=suffix, mode="ab") as temp_file:
       temp_file.write(response.read())
       yield temp_file
```

But Matt is way smarter than me. If he says this is a bad idea, I believe him :)

Reply:
I'm not familiar with the difference between urllib.request.urlopen and urllib.request.urlretrieve

I do know that after writing to a temp file, you have to close it before trying to do more -> this is what the "multi_access_temp_file" is trying to provide, create a temp file, close it (so that it's not "open" and thread locked), then cleanup once exiting the context manager.

Reply:
Oooh, closing it so it isn't locked makes sense. I think my suggestion probably breaks that. I'll drop.